ocaml/typing/printtyp.ml

```diff
@@ -1434,11 +1444,20 @@ and tree_of_typ_gf (ty, gf) =
   in
   (tree_of_typexp Type Alloc.Const.legacy ty, gf)
 
-and tree_of_ret_typ mode acc_mode (m, ty) =
+(** We are on the RHS of an arrow type, where [ty] is the return type, and [m]
+    is the return mode. This function decides the printed modes on [ty].
+    - If [ty] is another arrow type, [acc_mode] is the mode that has accumulated
+      from the currying, and thus the mode that the user would interpret as on
+      [ty] if it doesn't have parens around it.
+    - If [ty] is not an arrow type, [acc_mode] is meaningless.
+
+    NB: This function might mutate states; the caller is responsible for
+    reverting them. *)
+and tree_of_ret_typ_unsafe mode acc_mode (m, ty) =
```

Comment:
I really dislike using `unsafe` to mean so many different things. (That is, different uses of it mean different things.) How about `effectful`? `mutating`? `impure`? `with_effects`? `should_be_backtracked`? Any of these would be better than `unsafe` for me.

Reply:
Good point - changed to `mutating`.